### DISCUSSION TITLE
v0.29.0

### DISCUSSION THREAD

**Discussion by alimd** [AliMD/alwatr#796] [Announcements]
## What's Changed

* fix: compatible with new esbuild api by @AliMD in https://github.com/AliMD/alwatr/pull/695
* feat(surface)!: new surface element in ui-kit, remove old SurfaceElement class and add active-outline state by @AliMD in https://github.com/AliMD/alwatr/pull/696
* feat(ui/style): mobile only style by @AliMD in https://github.com/AliMD/alwatr/pull/697
* feat(ui-kit): simple button element by @AliMD in https://github.com/AliMD/alwatr/pull/698
* refactor(icon-button)!: rename ols standard-icon-button element by @AliMD in https://github.com/AliMD/alwatr/pull/699
* feat(icon-box): ass slot, highlight, pre-line, enhance typescale, enhance template by @AliMD in https://github.com/AliMD/alwatr/pull/700
* feat(ui-kit): text-field element by @AliMD in https://github.com/AliMD/alwatr/pull/701
* fix(ui-kit/chat): compatibility with new icon-button by @AliMD in https://github.com/AliMD/alwatr/pull/702
* fix: demo-pwa and ui-kit surface element file name and imports by @AliMD in https://github.com/AliMD/alwatr/pull/703
* feat(soffit-pwa): page form by @njfamirm in https://github.com/AliMD/alwatr/pull/684
* feat(soffit-form): submit form by @njfamirm in https://github.com/AliMD/alwatr/pull/693
* feat(soffit/form): expand/collapse animation by @AliMD in https://github.com/AliMD/alwatr/pull/705
* feat(soffit/form): improve animation and clear extra styles by @AliMD in https://github.com/AliMD/alwatr/pull/707
* feat(soffit/form): enhance animation and style by @AliMD in https://github.com/AliMD/alwatr/pull/709
* feat(soffit-pwa): form validator and simple message by @AliMD in https://github.com/AliMD/alwatr/pull/710
* feat(cloud): soffit form deploy by @njfamirm in https://github.com/AliMD/alwatr/pull/704
* fix(soffit-pwa): server deploy issues by @njfamirm in https://github.com/AliMD/alwatr/pull/711
* refactor: clean scripts by @AliMD in https://github.com/AliMD/alwatr/pull/725AliMD/alwatr/pull/715alwatr/pull/713
* feat(signal)!: new signal api and interface by @AliMD in https://github.com/AliMD/alwatr/pull/720
* feat(fetch): add update_cache to CacheStrategy by @AliMD in https://github.com/AliMD/alwatr/pull/733
* feat(i18n)!: new i18n from scratch! by @AliMD in https://github.com/AliMD/alwatr/pull/732
* feat(i18n): review by @njfamirm in https://github.com/AliMD/alwatr/pull/734
* feat(router): new api from scratch! by @AliMD in https://github.com/AliMD/alwatr/pull/735
* feat(alwatr-util): new core package by @njfamirm in https://github.com/AliMD/alwatr/pull/717
* fix: compatible with new api's and build issues by @AliMD in https://github.com/AliMD/alwatr/pull/736
* feat(util): untilEvent types, refactor to arrow function by @AliMD in https://github.com/AliMD/alwatr/pull/739
* feat(element): rewrite all mixins with new api by @AliMD in https://github.com/AliMD/alwatr/pull/740
* feat(soffit-pwa): form close animation by @AliMD in https://github.com/AliMD/alwatr/pull/741
* fix: clean ff old packages by @AliMD in https://github.com/AliMD/alwatr/pull/742
* feat(element/direction): compute dir mode and improve performance by @AliMD in https://github.com/AliMD/alwatr/pull/743
* chore(yarn): clean and upgrade by @AliMD in https://github.com/AliMD/alwatr/pull/744
* feat(snackbar): new ui-kit snackbar element by @njfamirm in https://github.com/AliMD/alwatr/pull/719
* feat(signal): separate request command with response by @AliMD in https://github.com/AliMD/alwatr/pull/750
* fix(ui): some visual issues by @AliMD in https://github.com/AliMD/alwatr/pull/751
* feat(snackbar): command signal and controller by @AliMD in https://github.com/AliMD/alwatr/pull/753
* feat(pwa-helper): new package by @AliMD in https://github.com/AliMD/alwatr/pull/754
* feat(pwa-helper): get config by @njfamirm in https://github.com/AliMD/alwatr/pull/759alwatr/pull/747AliMD/alwatr/pull/748
* feat(soffit-pwa): new card by @njfamirm in https://github.com/AliMD/alwatr/pull/752
* feat(i18n): add replaceNumber and auto detect setLocale from html by @AliMD in https://github.com/AliMD/alwatr/pull/764
* feat(spffir-pwa): service worker signals and user snackbar message on events by @AliMD in https://github.com/AliMD/alwatr/pull/765
* feat(nginx): optimize for pwa by @njfamirm in https://github.com/AliMD/alwatr/pull/766
* feat(pwa, i18n): dynamic loader by @AliMD in https://github.com/AliMD/alwatr/pull/767
* fix(github-actions): specific all versions by @AliMD in https://github.com/AliMD/alwatr/pull/770
* feat(math): uuid by @njfamirm in https://github.com/AliMD/alwatr/pull/773772
* feat(element): Alwatr logger index by @AliMD in https://github.com/AliMD/alwatr/pull/776
* feat(soffit-pwa): new version with i18n, content loader, form submit signal by @njfamirm in https://github.com/AliMD/alwatr/pull/769
* feat(form-registration): remoteAddress and deviceId by @njfamirm in https://github.com/AliMD/alwatr/pull/775
* fix(element): logger mixin issue by @njfamirm in https://github.com/AliMD/alwatr/pull/781
* feat(icon): enhance preload, support `icon-url-prefix` context signal by @njfamirm in https://github.com/AliMD/alwatr/pull/782AliMD/alwatr/pull/785pull/778alwatr/pull/777
* feat(icon-button): new standard icon button by @AliMD in https://github.com/AliMD/alwatr/pull/786
* fix(typescript): rollback to v4.9.5 by @AliMD in https://github.com/AliMD/alwatr/pull/787
* feat(icon-button): new content prop with click signal event by @AliMD in https://github.com/AliMD/alwatr/pull/788
* feat(top-app-bar): new `alwatr-top-app-bar` element by @MM25Zamanian in https://github.com/AliMD/alwatr/pull/623
* feat(soffit): home page top app bar by @AliMD in https://github.com/AliMD/alwatr/pull/789
* feat(soffit, snackbar): enhance spacing by @AliMD in https://github.com/AliMD/alwatr/pull/790
* feat(ui-kit/image-box): new element by @njfamirm in https://github.com/AliMD/alwatr/pull/721alwatr/pull/792
* feat(com-api): routes and refactor all services types by @njfamirm in https://github.com/AliMD/alwatr/pull/627

* chore(deps-dev): bump esbuild from 0.17.3 to 0.17.4 by @dependabot in https://github.com/AliMD/alwatr/pull/708
* chore(deps): bump tslib from 2.4.1 to 2.5.0 by @dependabot in https://github.com/AliMD/alwatr/pull/723
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.48.2 to 5.49.0 by @dependabot in https://github.com/
* chore(deps-dev): bump @typescript-eslint/parser from 5.48.2 to 5.49.0 by @dependabot in https://github.com/AliMD/
* chore(deps-dev): bump eslint from 8.32.0 to 8.33.0 by @dependabot in https://github.com/AliMD/alwatr/pull/738
* chore(deps-dev): bump esbuild from 0.17.4 to 0.17.5 by @dependabot in https://github.com/AliMD/alwatr/pull/737
* chore(deps-dev): bump @typescript-eslint/parser from 5.49.0 to 5.50.0 by @dependabot in https://github.com/AliMD/
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.49.0 to 5.50.0 by @dependabot in https://github.com/
* chore(deps): bump docker/build-push-action from 3 to 4 by @dependabot in https://github.com/AliMD/alwatr/pull/749
* chore(deps-dev): bump @types/node from 18.11.18 to 18.11.19 by @dependabot in https://github.com/AliMD/alwatr/pull/
* chore(deps-dev): bump esbuild from 0.17.5 to 0.17.6 by @dependabot in https://github.com/AliMD/alwatr/pull/780
* chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.50.0 to 5.51.0 by @dependabot in https://github.com/
* chore(deps-dev): bump @types/node from 18.11.19 to 18.13.0 by @dependabot in https://github.com/AliMD/alwatr/pull/783
* chore(deps): bump github/codeql-action from 2.2.1 to 2.2.2 by @dependabot in https://github.com/AliMD/alwatr/pull/779
* chore(deps): bump docker/setup-buildx-action from 2.4.0 to 2.4.1 by @dependabot in https://github.com/AliMD/alwatr/
* chore(deps-dev): bump @typescript-eslint/parser from 5.50.0 to 5.51.0 by @dependabot in https://github.com/AliMD/
* chore(deps-dev): bump peepeettier from 2.8.3 to 2.8.4 by @dependabot in https://github.com/AliMD/alwatr/pull/791
* chore(deps): bump github/codeql-action from 2.2.2 to 2.2.3 by @dependabot in https://github.com/AliMD/alwatr/pull/793
* chore(deps-dev): bump @material/web from 1.0.0-pre.1 to 1.0.0-pre.2 by @dependabot in https://github.com/AliMD/
* chore(deps-dev): bump esbuild from 0.17.6 to 0.17.7 by @dependabot in https://github.com/AliMD/alwatr/pull/794

**Full Changelog**: https://github.com/AliMD/alwatr/compare/v0.28.0...v0.29.0

<hr /><em>This discussion was created from the release <a href='https://github.com/AliMD/alwatr/releases/tag/v0.29.0'>v0.29.0</a>.</em>

**Comment by MM25Zamanian** [comment]
واقعا نمیدونم خدا چه دشمنی ای با من داره
هم تمام کد های لازم الوتر رو کپی کردم و پروژه رو پابلیش کردم ریلیز گرفتی 

دلم میخواد یکی رو پیدا کنم فقط بهش فحش بدم
